### PR TITLE
Fix combat testing tracker failures

### PR DESCRIPTION
**Pull request by zunath**
## Summary

Addresses every current `Fail` entry in the [combat-upgrade testing tracker](https://docs.google.com/spreadsheets/d/1iHMKtrnh3lbUnmrgXtxEQseAJd7WIL6RU51RVSktm4s/edit?gid=2101115203#gid=2101115203):

- replace Shadow Step I/II's weapon-swing animation with a non-attack cast-out animation
- keep Ghost Protocol discoverable when the primary quickbar is full by moving its fighter class-feat entry below the client scan limit
- describe Break Morale I/II using the player-facing `Ability Accuracy` stat and refresh the Design Bible/audit manifests/TLK assets
- snapshot Rousing Shout's danger threshold before temporary HP is applied, allowing ranks II/III to apply their stronger low-HP rescue reduction
- document and test that the stronger Bolster Resolve II reduction can legitimately produce the reported 15% rank-I result without stacking

## Validation

- `dotnet build SWLOR.Game.Server.Tests\SWLOR.Game.Server.Tests.csproj --no-restore -p:RunPostBuildEvent=Never` (passes; one existing nullable-context warning)
- 52 focused Espionage, Leadership, radial-menu, animation, and Design Bible sync tests pass
- `NPCEnemyBalanceAuditTests.AllNpcHpBudgets_AccountForNativeVitalityAndToughnessRules` passes
- regenerated TLK binary round-trip verifies entries 5345 and 5411
- `git diff --check` passes in the parent and Haks repositories

## Known baseline failure

`CombatUpgradeBibleManifest_MatchesLivePerkAndAbilityRegistries` still reports three unrelated pre-existing status-effect class/display-name mismatches (`RestoredFPForceAttackStatusEffect`, `RestoredFPHasteStatusEffect`, and `RestoredStaminaAttackStatusEffect`). None of those files are changed by this PR; all other selected Design Bible sync tests pass.

## Companion PR

- zunath/SWLOR_Haks#385

## Summary by CodeRabbit

* **New Features**
  * Shadow Step now uses an updated casting animation.
  * Break Morale descriptions now accurately describe reduced Ability Accuracy.
* **Bug Fixes**
  * Rousing Shout now determines low-health rescue effects before granting temporary hit points.
  * Leadership damage-reduction effects now reconcile correctly, preventing weaker effects from stacking.
* **Tests**
  * Expanded coverage for low-health ability behavior, radial-menu placement, animations, status effects, and ability descriptions.

### CODE REVIEW

**Comment by zunath** [comment]
@codex review for regressions, missing tests, and risky behavior changes. Please apply the repository AGENTS.md rules and review the linked Haks companion PR where relevant.

**Comment by zunath** [comment]
@codex review

Please perform a fresh full review of exact head `dfa4a6a7b0bc2776213d0c5eb5aebc9dcdadf05f` against `feature/combat-upgrade`, applying the repository AGENTS.md rules and considering companion Haks PR #385. Report every actionable correctness, regression, data-integrity, documentation-sync, or missing-test issue; if none remain, state that explicitly.

File: SWLOR.Game.Server.Tests/Perks/LeadershipCombatUpgradeTests.cs [hunk 1]
```diff
@@ -106,6 +106,15 @@ public void ReportedLeadershipFailPerks_UseDynamicRadiusAndTemporaryHitPoints()
         markTargetRider.Should().BeGreaterThan(affectedTargetGuard,
             "Break Morale must grant Mark Target once after at least one direct status application succeeds");
 
+        var rousingShout = File.ReadAllText((root / "SWLOR.Game.Server" / "Feature" / "AbilityDefinition" / "Leadership" / "RousingShoutAbilityDefinition.cs").FullName);
+        var dangerSnapshot = rousingShout.IndexOf("var targetWasInDanger = IsTargetInDanger(target);", StringComparison.Ordinal);
```
[the remark this answers is not in the document]
Comment (zunath) [reply SWLOR.Game.Server.Tests/Perks/LeadershipCombatUpgradeTests.cs:111]: Fixed in `22a3a0ab38`. The engine behavior harness now supports an explicit pre-activation target HP percentage. Rousing Shout I/II/III each execute at exactly 35% HP and assert their rank-specific status plus temporary HP through the live activation path. Build passes; Leadership and behavior-contract tests pass 16/16.

File: SWLOR.Game.Server.Tests/Perks/LeadershipCombatUpgradeTests.cs [hunk 2]
```diff
@@ -421,11 +442,11 @@ public void LeadershipFeatAndAbilityDescriptions_MatchCombatBible()
             (FeatType.ChargeOrder1, "Aura: Party members within Leadership range (5m base) gain +10% movement speed and +30 Mobility Resistance. SOC scaling can raise these to +12% movement speed and +40 Mobility Resistance."),
             (FeatType.PressTheAttack2, "Party members within Leadership range (5m base) deal +8% damage for 30 seconds. SOC scaling can raise this to +10%."),
             (FeatType.RallyingStandard2, "Aura: Party members within Leadership range (5m base) gain +5% physical and Force ability hit chance. SOC scaling can raise this to +6%."),
-            (FeatType.BreakMorale1, "Enemies within Leadership range (5m base) suffer Flash for 30 seconds, reducing physical and Force ability hit chance by 10%. SOC scaling can raise the penalty to 12%. This command applies reliably to valid enemies within Leadership range (5m base)."),
+            (FeatType.BreakMorale1, "Enemies within Leadership range (5m base) suffer Flash for 30 seconds, reducing Ability Accuracy by 10%. SOC scaling can raise the penalty to 12%. This command applies reliably to valid enemies within Leadership range (5m base)."),
```
[the remark this answers is not in the document]
Comment (zunath) [reply SWLOR.Game.Server.Tests/Perks/LeadershipCombatUpgradeTests.cs:447]: Fixed in `22a3a0ab38`. `LeadershipFeatAndAbilityDescriptions_MatchCombatBible` now decodes the deployed `sw_tlk.tlk` with the repository's `TlkReader` and compares each feat/spell description entry to the JSON-backed expected text, including IDs 5345 and 5411. The focused 52-test suite passes.

File: SWLOR.Game.Server/Feature/PerkDefinition/LeadershipVanguardCommandPerkDefinition.cs [hunk 3]
```diff
@@ -143,13 +143,13 @@ private void BreakMorale()
                 .Name("Break Morale")
 
                 .AddPerkLevel()
-                .Description("Enemies within Leadership range (5m base) suffer Flash for 30 seconds, reducing physical and Force ability hit chance by 10%. SOC scaling can raise the penalty to 12%. This command applies reliably to valid enemies within Leadership range (5m base).")
+                .Description("Enemies within Leadership range (5m base) suffer Flash for 30 seconds, reducing Ability Accuracy by 10%. SOC scaling can raise the penalty to 12%. This command applies reliably to valid enemies within Leadership range (5m base).")
```
[the remark this answers is not in the document]
Comment (zunath) [reply SWLOR.Game.Server/Feature/PerkDefinition/LeadershipVanguardCommandPerkDefinition.cs:146]: Fixed in `22a3a0ab38`. Updated `design/testing/SWLOR CU Testing Tracker.xlsx` cells F495/F501 surgically to use Ability Accuracy, verified the exact cell values, and rendered the Leadership range to confirm the checked-in QA tracker remains intact.

**Comment by zunath** [comment]
All three findings from the first Codex pass are fixed in exact head `22a3a0ab38f4e5d352faca90356be11a9b23e916`, replied to with evidence, and resolved.

Validation on this head:
- build succeeds with `RunPostBuildEvent=Never`
- original focused gameplay/audit suite passes 52/52
- Leadership plus executable behavior-contract coverage passes 16/16
- checked-in QA tracker cells were verified and visually rendered
- deployed TLK binary is decoded and compared to the JSON source in the test suite
- `git diff --check` is clean and unresolved review threads are 0

@codex review

Please perform a fresh full review of exact head `22a3a0ab38f4e5d352faca90356be11a9b23e916` against `feature/combat-upgrade`, including all first-pass fixes and companion Haks PR #385. Report every remaining actionable issue; if none remain, state that explicitly.
